master/internal/templates/api.go

```diff
@@ -109,29 +109,56 @@ func (a *TemplateAPIServer) PutTemplate(
 	if len(req.Template.Name) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "name is required")
 	}
-	if req.Template.WorkspaceId != 0 {
-		return nil, status.Errorf(codes.InvalidArgument, "setting workspace_id is not supported.")
-	}
 
-	switch _, err := TemplateByName(ctx, req.Template.Name); {
+	switch tpl, err := TemplateByName(ctx, req.Template.Name); {
 	case errors.Is(err, db.ErrNotFound):
 		_, err := a.PostTemplate(ctx, &apiv1.PostTemplateRequest{Template: req.Template})
 		if err != nil {
 			return nil, err
 		}
+		return &apiv1.PutTemplateResponse{Template: req.Template}, nil
 	case err != nil:
 		return nil, err
 	default:
-		req := &apiv1.PatchTemplateConfigRequest{
-			TemplateName: req.Template.Name,
-			Config:       req.Template.Config,
+		user, _, err := grpcutil.GetUser(ctx)
+		if err != nil {
+			return nil, err
+		}
+		permErr, err := AuthZProvider.Get().CanUpdateTemplate(
+			ctx, user, model.AccessScopeID(tpl.WorkspaceID),
+		)
+		switch {
```

Comment:
Why use a switch instead of `if err != nil` here? (I'm wondering if it's a style convention I'm not familiar with?)

Reply:
This part was moved from other places, I saw we use both switch and if under similar circumstances.